iep-module-userservice/src/main/java/com/netflix/iep/userservice/CompositeUserService.java

```diff
@@ -46,7 +49,7 @@
   @Override public Set<String> emailAddresses() {
     Set<String> vs = new TreeSet<>();
     for (UserService service : services) {
-      vs.addAll(service.emailAddresses());
+      vs.addAll(service.emailAddresses().stream().map(String::toLowerCase).collect(toSet()));
```

Comment:
This would do the conversion work on each access, it would be better to do it early when the data is cached.